### PR TITLE
DISCUSSION: add support for meta get parser

### PR DESCRIPTION
**Pull request by danmayer**
### TL;DR - What are you trying to accomplish?

We want to have support for both the ascii protocol and meta protocol this is the first PR to move in that direction and lays the ground work for building out the meta protocol and returning the meta tag values back to the caller.

### Details - How are you making this change?  What are the effects of this change?
Related to: https://github.com/Shopify/async-memcached/pull/64

As a prototype step to break changes up into smaller chunks, we are adding a new struct `MetaValue` that is part of the `Value` struct response.  Going forward, this will be refactored so that the two structs are separated and the `Value` struct will retain its original shape.  The new `MetaValue` can contain optional flag response value for any requested flags. The parser will parse all implemented flag responses and it should be easy to extend to add support for additional meta flags.

### CODE REVIEW

**Review by mrattle — COMMENTED** [review]
Left some initial comments, which I have started prototyping around in another branch.

File: src/lib.rs [hunk 1]
```diff
@@ -353,6 +403,82 @@ impl Client {
...
+    // - <datalen> is the length of the payload data.
+    //
+    // - <flags> are a set of single character codes ended with a space or newline.
+    //   flags may have strings after the initial character.
+    pub async fn meta_set<K, V>(
+        &mut self,
+        key: K,
+        value: V,
+        ttl: Option<i64>,
+        meta_flags: FxHashMap<&[char], String>,
```
Comment (mrattle) [inline src/lib.rs:427]: Wondering if it makes more sense to take `meta_flags` input here as `&[char]`, same as we do for `meta_get`?  Ideally the parameter would be the same type across all meta methods to provide a cohesive experience for the user.

Alternatively, everything takes a HashMap of flags.  I understand the differentiation in implementations as-written today since currently we're only considering the flags that we need for probabilistic hotkey detection in `mg` which have no follow-on tokens, but when looking at the full spec we will need similar support between all meta methods.

**Comment by mrattle** [comment]
Closed in favour of: https://github.com/Shopify/async-memcached/pull/66, https://github.com/Shopify/async-memcached/pull/67, https://github.com/Shopify/async-memcached/pull/68
